guides/source/active_record_querying.md

```diff
@@ -1935,6 +1935,61 @@ SELECT books.* FROM books WHERE books.out_of_print = true
 
 [`unscoped`]: https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Default/ClassMethods.html#method-i-unscoped
 
+### New Chains Inside Scope Block
+
+Unlike class methods, [`scope`][] can easily start a new clean chain against the
+model it is defined on.
+
+```ruby
+class Topic < ApplicationRecord
+  scope :toplevel, -> { where(parent_id: nil) }
+  scope :children, -> { where.not(parent_id: nil) }
+  scope :has_children, -> {
+    where(id: Topic.children.select(:parent_id))
+  }
+end
+
+Topic.toplevel.has_children
+```
+
+Inside `has_children` the `Topic` chain generates a subquery like this:
+
+```sql
+SELECT "topics"."parent_id" FROM "topics" WHERE "topics"."parent_id" IS NOT NULL
+```
+
+Class methods have different behavior which can be surprising if you expect them
```

Comment:
Why are we documenting it is surprising? Either it is documented behavior or it is surprising.

Reply:
I'll shorten it to just "Class methods have different behavior."